src/components/maincontroller.ts

```diff
@@ -194,33 +203,30 @@ window.mediaManager.addEventListener(
     }
 );
 
-console.log('Application is ready, starting system');
-
-export function reportDeviceCapabilities() {
-    getMaxBitrate().then((maxBitrate) => {
-        let capabilitiesUrl = JellyfinApi.createUrl(
-            'Sessions/Capabilities/Full'
-        );
-
-        let deviceProfile = getDeviceProfile(maxBitrate);
+export function reportDeviceCapabilities(): Promise<void> {
+    return getMaxBitrate().then((maxBitrate) => {
+        const deviceProfile = getDeviceProfile({
+            enableHls: true,
+            bitrateSetting: maxBitrate
+        });
 
-        let capabilities = {
+        const capabilities = {
             PlayableMediaTypes: ['Audio', 'Video'],
             SupportsPersistentIdentifier: false,
             SupportsMediaControl: true,
             DeviceProfile: deviceProfile
         };
-        window.hasReportedCapabilities = true;
+        hasReportedCapabilities = true;
 
-        return JellyfinApi.authAjax(capabilitiesUrl, {
+        return JellyfinApi.authAjax('Sessions/Capabilities/Full', {
             type: 'POST',
             data: JSON.stringify(capabilities),
             contentType: 'application/json'
         });
     });
 }
 
-export function processMessage(data) {
+export function processMessage(data: any): void {
```

Comment:
Same as above

Reply:
This one will need a new interface I believe